cli/azd/pkg/spin/spin.go

```diff
@@ -126,10 +124,10 @@ func GetSpinner(ctx context.Context) *Spinner {
 
 // Gets a spinner from the specified context, otherwise creates a new instance
 // Returns a new context when a new spinner is created
-func GetOrCreateSpinner(ctx context.Context, title string) (*Spinner, context.Context) {
+func GetOrCreateSpinner(ctx context.Context, title string, writer io.Writer) (*Spinner, context.Context) {
```

Comment:
I'm a little confused between the io.Writter and the `console`.  Can we use the `console` component we have everywhere?  (instead of io.Writter) ?  

the console providers the in and out capabilities...  so, in cases where, in the future we would need to have `input`, and the io.Writter wouldn't be enough, the console seems like a better option.

Reply:
I agree there is a lot of overlap between these two types.  I thought about adding something like a `Writer()` property on the `Console` interface so I could take a Console here instead, but ran into the problem of figuring out how to implement it in a sane way for `MockConsole` which made me realize maybe I should keep the types separate.

My main goal has been to make this previously hidden dependency visible - I will think more about the right way to flow it.